src/install/dracut-install.c

```diff
@@ -1667,6 +1668,11 @@ static int install_dependent_modules(struct kmod_ctx *ctx, struct kmod_list *mod
         const char *supplier_path;
         Iterator i;
         HASHMAP_FOREACH(supplier_path, suppliers_paths, i) {
+                if (check_hashmap(processed_suppliers, supplier_path))
+                        continue;
+
+                hashmap_put(processed_suppliers, strdup(supplier_path), strdup(supplier_path));
```

Comment:
Couldn't this 2 `strdup` calls be simplified to just one using a temporary `char*` variable?

```c
char *path = strdup(supplier_path);
hashmap_put(processed_suppliers, path, path);
```

Other than that, your patch looks good, thanks.

Reply:
Fixed, thank you!